File: robertly-net-api/api/Models/Goal.cs
using System;

namespace robertly.Models;

public record Goal
{
  public int GoalId { get; set; }
  public int UserId { get; set; }
  public string? GoalType { get; set; }
  public decimal TargetValue { get; set; }
  public DateTime? TargetDate { get; set; }
  public int? ExerciseId { get; set; }
  public string? MuscleGroup { get; set; }
  public DateTime CreatedAtUtc { get; set; }
}

File: robertly-ui/src/app/models/goal.ts
export interface Goal {
  goalId?: number;
  userId?: number;
  goalType: 'bodyweight' | 'series-per-muscle' | 'exercise-personal-record' | 'calories' | 'protein';
  targetValue: number;
  targetDate?: string;
  exerciseId?: number;
  muscleGroup?: string;
  createdAtUtc?: string;
}